File: 3634. Minimum Removals to Balance Array/minRemoval.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int minRemoval(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int start = 0;
        for(int end = 0 ; end<nums.size(); end++){
            if(nums[end] > 1LL*nums[start]*k){
                start++;
            }
        }
        return start;
    }
};

File: 3634. Minimum Removals to Balance Array/minRemoval.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minRemoval = function (nums, k) {
    nums.sort((a, b) => a - b)
    let start = 0

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] > nums[start] * k) {
            start++
        }
    }
    return start;


};